.github/CODE_STANDARDS.md

```diff
@@ -0,0 +1,295 @@
+  - [4. Documentation Code Standards](#4-documentation-code-standards)
+    - [4.1 Overview](#41-overview)
+    - [4.2 Principles](#42-principles)
+    - [4.3 Feature Documentation](#43-feature-documentation)
+      - [4.3.1 Necessity](#431-necessity)
+      - [4.3.2 Examples](#432-examples)
+    - [4.4 External Features](#44-external-features)
+    - [4.5 Markdown usage](#45-markdown-usage)
+    - [4.5 Conclusion](#45-conclusion)
+
+## Most important rules - Quick Checklist
+
+- [ ] Follow official Go guidelines for style and formatting.
+- [ ] Write self-documenting code and minimize comments.
+- [ ] Ensure comprehensive test coverage including happy and error paths.
+- [ ] Provide meaningful and constructive code reviews.
+- [ ] Adhere to Conventional Commits for commit messages and PR naming.
+- [ ] Document every feature adequately, especially for open-source projects.
+- [ ] Keep documentation clear, concise, up-to-date, and accessible.
+- [ ] Branching - choose consistent naming conventions, include issue number, delete branches after merging.
+
+## 1. Code style and formatting - official guidelines
+
+### 1.1 In GO applications or libraries, we follow the official guidelines
+
+- [Effective Go](https://go.dev/doc/effective_go) - official Go guidelines
+- [Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments) - official Go code review comments
+- [Go Examples](https://pkg.go.dev/testing#hdr-Examples) - official Go examples - used in libraries to explain how to use their exposed features
+- [Go Test](https://pkg.go.dev/testing) - official Go testing package & recommendations
+- [Go Linter](https://golangci-lint.run/) - golangci-lint - only codestyle checks
+
+ > Our current linter configuration is in the `.golangci.yml` file.
+
+#### Additional useful resources with GO recommendations, best practices and the common mistakes
+
+- [Go Styles by Google](https://google.github.io/styleguide/go/) - Google's Go Style Guide
+- [Uber Go Style Guide](https://github.com/uber-go/guide/blob/master/style.md) - Uber's Go Style Guide
+- [Go Common Mistakes](https://github.com/golang/go/wiki/CommonMistakes) - Common Mistakes in Go
+
+## 2. Code Rules
+
+### 2.1. Self-documenting code
+
+#### As a Developer
+
+- Refactoring tasks should be strategically undertaken; not every piece of code warrants modification. If a segment of code, despite comments, remains untouched and unproblematic, it is   likely fulfilling its purpose effectively.
+- When embarking on feature additions or bug fixes, and encountering commented code, consider extracting functions and creating a PR before proceeding with the primary task. This helps maintain code clarity and function.
+- Should you come across a superfluous comment during your changes, do take the initiative to remove it and create a PR for that. This practice contributes to keeping the repository neat and well-maintained.
```

Comment:
I'm not sure if the sentence is written correctly
```suggestion
- If you come across a superfluous comment during your changes, take the initiative to remove it and create a PR for that. This practice contributes to keeping the repository neat and well-maintained.
```

Reply:
It is correct, a bit more formal and sublime tho :)